dvc/commands/repro.py

```diff
@@ -55,7 +55,16 @@ def add_arguments(repro_parser):
     repro_parser.add_argument(
         "targets",
         nargs="*",
-        help="Stages to reproduce. 'dvc.yaml' by default.",
+        help=(
+            """\
+Stages to reproduce. 'dvc.yaml' by default.
+The targets can be path to a dvc.yaml file or `.dvc` file,
+or a stage name or a foreach-group name from dvc.yaml file from
+current working directory. To run a stage/foreach-group from dvc.yaml
+from other directories, the target must be a path followed by colon `:`
+and then the stage name or foreach-group name.
```

Comment:
Noticing this longer text (without going into whether it's needed in the help output or not), keep in mind that there are tools like spell checkers and Grammarly to help fix syntax. Beyond that we have a larger group of people you can request sign offs for these (docs, devrel, cse) so please consider incorporating this into your process cc @dberenbaum thanks!